src/@types/CommandContext.d.ts

```diff
@@ -0,0 +1,103 @@
+import type {
+	Client,
+	CommandInteraction,
+	CommandInteractionOption,
+	DMChannel,
+	Guild,
+	GuildMember,
+	GuildTextBasedChannel,
+	InteractionReplyOptions,
+	Message,
+	MessageReplyOptions,
+	User,
+} from 'discord.js';
+
+declare global {
+	/** Information relevant to a command invocation. */
+	interface CommandContext {
+		/** The command invocation interaction. */
+		readonly interaction: CommandInteraction;
+
+		/** Our own signed-in Discord client. */
+		readonly client: Client<true>;
+
+		/** The place to submit informative debug messages. */
+		readonly logger: Console; // TODO: More extensible logger than `console`?
+
+		/** The guild in which the command was invoked. */
+		readonly guild: Guild | null;
```

Comment:
Only that `Guild?` is not valid TypeScript syntax

Reply:
EDIT: I seem to have been mistaken. TIL that `Guild?` means `Guild | null`.

Reply:
I think (for now at least) I'd like to leave `| null` suffixes, so that my monkey brain down the line doesn't confuse `Foo?` for `Foo | undefined`, when it really means `Foo | null`.